go.go

```diff
@@ -99,35 +99,55 @@ func Go(a, b []byte) (changes []*Change, err error) {
 			ds := x.diffstat(ai, bi)
 			changes = append(changes, &Change{Name: s, DelLines: ds.del, InsLines: ds.ins})
 		} else {
-			changes = append(changes, &Change{Name: s, Inserted: true, InsLines: len(bytes.Split(x.bBytes(bi), newline))})
+			changes = append(changes, &Change{Name: s, Inserted: true, InsLines: x.bLines(bi)})
 		}
 	}
 	for s, ai := range del {
 		if _, ok := ins[s]; !ok {
-			changes = append(changes, &Change{Name: s, Deleted: true, DelLines: len(bytes.Split(x.aBytes(ai), newline))})
+			changes = append(changes, &Change{Name: s, Deleted: true, DelLines: x.aLines(ai)})
 		}
 	}
 
 	if other != nil {
-		// Calculate an overall diffstat.
-		aLines := bytes.Split(a, newline)
-		bLines := bytes.Split(b, newline)
-		ab := diff.Bytes(aLines, bLines)
-		es := diff.Myers(context.Background(), ab)
-		ins, del := es.Stat()
-		// Subtract all diffs accountable for by other changes.
-		for _, c := range changes {
-			ins -= c.InsLines
-			del -= c.DelLines
+		// Calculate diffstats between non-declarations.
+		aIsLarger := false
+		lens := len(x.asplit) - 1
+		if len(x.asplit) > len(x.bsplit) {
+			lens = len(x.bsplit) - 1
+			aIsLarger = true
 		}
-		if ins < 0 {
-			ins = 0
+		// If df is increased, it means that an index has been displaced by some change.
+		var df int64
```

Comment:
Why not just int? Then you can remove some casts below. I'm really not worried about a file having > 2^31 lines on a 32 bit machine.

Reply:
It was for accommodating the needs of the abs function, but since the abs function has been simplified to work with ints, df can be changed to int and the int64 castings removed.